GitCommands/Git/GitModule.cs

```diff
@@ -2202,7 +2202,7 @@ public string AddRemote(string? name, string? path)
         public IReadOnlyList<string> GetRemoteNames()
         {
             return _gitExecutable
-                .Execute("remote")
+                .Execute("remote", cache: null) //// GitCommandCache is not cleared when switching repos
```

Comment:
cache is null by default, must be explicitly added

Reply:
I have created this very experimental WIP PR such early mainly for the demonstration of the issue #9865 if using:
```suggestion
                .Execute("remote", cache: GitCommandCache)
```
I should have mentioned this in the issue description.

Reply:
Continuing this in #9865